### PR TITLE
Add mock httpClient for image tests

### PR DESCRIPTION
**Pull request by xqwzts**
Creates a mock http.Client that always returns StatusCode 200 and an image [taken from https://github.com/flutter/flutter/blob/master/dev/manual_tests/test/mock_image_http.dart] and uses it as the httpClient for image tests.

More info here: https://github.com/flutter/flutter/issues/13433

### CODE REVIEW

File: test/flutter_markdown_test.dart [hunk 1]
```diff
@@ -293,3 +300,19 @@ Widget _boilerplate(Widget child) {
...
+
+// Returns a mock HTTP client that responds with an image to all requests.
+ValueGetter<http.Client> createMockImageHttpClient = () {
+  return new http.MockClient((http.BaseRequest request) {
+    return new Future<http.Response>.value(
+        new http.Response.bytes(_kTransparentImage, 200, request: request));
+  });
+};
+
+const List<int> _kTransparentImage = const <int>[
```
Comment (DaveShuckerow) [inline test/flutter_markdown_test.dart]: The 'k' isn't necessary for constant names.  Go ahead and drop that: `const _transparentImage`

File: test/flutter_markdown_test.dart [hunk 2]
```diff
@@ -165,6 +168,8 @@ void main() {
   });
   
   testWidgets('Image links', (WidgetTester tester) async {
+    createHttpClient = createMockImageHttpClient;
```
Comment (DaveShuckerow) [inline test/flutter_markdown_test.dart]: Suggestion:

Move these into a group and prepare the image client in setUpAll():

```dart
group('Images', () {
  setUpAll(() {
    createHttpClient = ...;
  });
  testWidgets('should work with a link', ...);
  testWidgets('should show properly next to text', ...);
});
```

The style for naming test groups is for the group name to make a complete sentence when joined with each test's name.

Comment (xqwzts) [reply test/flutter_markdown_test.dart]: Done.

File: test/flutter_markdown_test.dart (hunk 1 above)
Comment (xqwzts) [reply test/flutter_markdown_test.dart]: Done

**Comment by xqwzts** [comment]
Updated.

I also changed the test names for the links group to match the sentence style.

**Pull request merged**